### PR TITLE
provide utility helpers for preserving internal annotations

### PR DESCRIPTION
**Pull request by natasha41575**
Task 4 in https://github.com/kubernetes-sigs/kustomize/issues/4024, implementing [function spec v1](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md#krm-functions-specification).

This adds some helper functions for function authors to kyaml for preserving annotations with the prefix `internal.config.kubernetes.io`. 

It should go in the same release as #4248 and #4249.

ALLOW_MODULE_SPAN

### CODE REVIEW

**Comment by natasha41575** [comment]
/hold

Making sure everyone can get their review in before this is merged.

File: kyaml/kio/kioutil/kioutil.go
Comment (natasha41575) [reply kyaml/kio/kioutil/kioutil.go]: fixed

File: kyaml/kio/kioutil/kioutil.go
Comment (natasha41575) [reply kyaml/kio/kioutil/kioutil.go]: done!

**Review by mengqiy — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
